File: Deep-Learning-for-Beginners-master/Python/ch.6/Pool.py
import numpy as np
from scipy import signal

    
def Pool(x):
    (xrow, xcol, numFilters) = x.shape
    y = np.zeros((int(xrow/2), int(xcol/2), numFilters))
    
    for k in range(numFilters):
        filter = np.ones((2,2)) / (2*2)
        image  = signal.convolve2d(x[:, :, k], filter, 'valid')
        
        y[:, :, k] = image[::2, ::2]

    return y

File: Deep-Learning-for-Beginners-master/ch.6/Pool.m
function y = Pool(x)
%     
% 2x2 mean pooling
%
%
[xrow, xcol, numFilters] = size(x);

y = zeros(xrow/2, xcol/2, numFilters);
for k = 1:numFilters
  filter = ones(2) / (2*2);    % for mean    
  image  = conv2(x(:, :, k), filter, 'valid');
  
  y(:, :, k) = image(1:2:end, 1:2:end);
end

end
 